apps/prairielearn/src/lib/timezones.ts

```diff
@@ -18,13 +18,36 @@ async function getAvailableTimezonesFromDB(): Promise<Timezone[]> {
   return availableTimezones;
 }
 
+/**
+ * Returns a list of all timezones supported by the database. The list is
+ * memoized so that, if it is needed more than once in the same session, the
+ * same list is returned.
+ */
 export async function getAvailableTimezones(): Promise<Timezone[]> {
   if (memoizedAvailableTimezones == null) {
     memoizedAvailableTimezones = await getAvailableTimezonesFromDB();
   }
   return memoizedAvailableTimezones;
 }
 
+/**
+ * Returns a filtered list of canonical timezones that are supported by both
+ * Postgres and the JavaScript Intl API. As per the specification of
+ * Intl.supportedValuesOf('timeZone'), the list includes only canonical timezone
+ * names, and does not include aliases or deprecated names.
+ *
+ * @param alwaysInclude - Optional array of timezone names to always include in
+ * the result, even if they're not canonical. These timezones are only included
+ * if they're supported by Postgres.
+ */
+export async function getCanonicalTimezones(alwaysInclude?: string[]): Promise<Timezone[]> {
+  const availableTimezones = await getAvailableTimezones();
+  const canonicalTimezones = Intl.supportedValuesOf('timeZone');
```

Comment:
This returns neither `Etc/UTC` (canonical) or `UTC` (link), but I tested both those locally and they still work with `Intl.DateTimeFormat`. This makes me nervous that there are others that are for some reason missing from `supportedValuesOf`.

Reply:
Turns out V8 is just being weird:

- https://github.com/tc39/ecma402/issues/778
- https://issues.chromium.org/issues/42203023

Seems like only Firefox has the expected behavior right now. I'd be inclined to manually include `UTC` in the list; note that in JS, the other variants of UTC are canonicalized to `UTC` "for historical reasons" (https://tc39.es/ecma402/#sec-use-of-iana-time-zone-database).